### PR TITLE
bugfix-refetching-deleted-runs

### PR DESCRIPTION
**Pull request by dylanmartin**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our [guidelines](../CONTRIBUTING.md)
- [x] Passes e2e testing
- [x] Passes lint

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
bugfix

* **What is the current behavior?** (You can also link to an open issue here)
closes _issue #1266 
cloese_issue #1240

* **What is the new behavior (if this is a feature change)?**
The ui stops fetching run status on deleted runs

### CODE REVIEW

File: packages/coinstac-api-server/src/data/resolvers.js [hunk 1]
```diff
@@ -1877,7 +1878,7 @@ const resolvers = {
     runWithHeadlessClientStarted: {
       subscribe: withFilter(
         () => pubsub.asyncIterator('runWithHeadlessClientStarted'),
-        (payload, variables) => (variables.clientId && variables.clientId in payload.runWithHeadlessClientStarted.clients)
+        (payload, variables) => (variables.clientId && variables.clientId in payload.runWithHeadlessClientStarted.pipelineSnapshot.headlessMembers)
```
Comment (rochaeb) [inline packages/coinstac-api-server/src/data/resolvers.js]: I think the the other way is the right one (using `.clients`)

File: packages/coinstac-api-server/src/data/resolvers.js [hunk 2]
```diff
@@ -532,7 +532,8 @@ const resolvers = {
       }
 
       if (!run) {
-        return Boom.notFound('Run not found');
+        // return Boom.notFound('Run not found');
```
Comment (rochaeb) [inline packages/coinstac-api-server/src/data/resolvers.js]: remove the comment

File: packages/coinstac-ui/app/render/components/runs/top-notification-progress-bar/index.jsx [hunk 3]
```diff
@@ -53,6 +53,10 @@ function TopNotificationProgressBar({ runs, consortia, router }) {
 
   // Check status of decentralized run
   useEffect(() => {
+    if (runStatus === null) {
+      setRun(null);
+    }
+
     if (!runStatus) return;
```
Comment (rochaeb) [inline packages/coinstac-ui/app/render/components/runs/top-notification-progress-bar/index.jsx]: You could set the run to null here. And return after.

**Review by rochaeb — CHANGES_REQUESTED** [review]

File: packages/coinstac-api-server/src/data/resolvers.js (hunk 1 above)
Comment (dylanmartin) [reply packages/coinstac-api-server/src/data/resolvers.js]: Yeah I don't know how that change even got there. Not something I did intentionally.

**Pull request merged**